nhs/data/join_on_sa1.py

```diff
@@ -0,0 +1,219 @@
+from typing import Any, List  # Importing required types
+
+import fiona
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import polars as pl
+import yaml
+from fiona.drvsupport import supported_drivers
+from loguru import logger
+from pathos.multiprocessing import ProcessingPool as Pool
+from pyproj import CRS
+
+# Set the SHAPE_RESTORE_SHX option
+supported_drivers["ESRI Shapefile"] = "rw"
+
+
+def read_config(config_path: str) -> dict:
+    """Read YAML configuration file."""
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+    logger.info(f"Configuration loaded: {config}")
+    return config
+
+
+def read_shapefile(path: str) -> int:
+    """Read shapefile and return total number of rows."""
+    total_rows = len(pl.read_parquet(path))
+    logger.info(f"Total number of rows in the shapefile: {total_rows}")
+    return total_rows
+
+
+def read_csv(path: str) -> pl.DataFrame:
+    """Read CSV file and return as Polars DataFrame."""
+    df = pl.read_csv(path)
+    logger.info(df.head())
+    return df
+
+
+def prepare_points(df: pl.DataFrame, crs: str) -> Any:
+    """Prepare points data for geospatial analysis."""
+    if "LONGITUDE" in df.columns and "LATITUDE" in df.columns:
+        df = df.rename({"LONGITUDE": "x", "LATITUDE": "y"})
+
+    gdf = gpd.GeoDataFrame(
+        df.to_pandas(),
+        geometry=gpd.points_from_xy(df["x"], df["y"]),
+        crs="EPSG:7844",  # Assuming input is in WGS84
+    )  # type: ignore
+    projected_crs = CRS.from_string(crs)
+    return gdf.to_crs(projected_crs)
+
+
+def process_chunk(map_data: Any, chunk: Any) -> pl.DataFrame:
+    """Process a chunk of point data, joining with area data."""
+    pnts_with_area = gpd.sjoin(chunk, map_data, how="left", predicate="within")
+    missing_points = pnts_with_area[pnts_with_area["SA2_NAME21"].isnull()]
```

Comment:
SA2_NAME21 is a column for the areas in map_data...? 

Is it correct interpretation to say we are making sure atleast one area code is being assigned.

Reply:
You can make `"SA2_NAME21"` an input to the function as well as I'm not sure if this is hard coding the column name. Just set the default value to `"SA2_NAME21"` in the parameter so we both hard-code but also allow the user to change the column name if needed

Reply:
Done, changes implemented in configurations.yml. Docstring added too.